pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -395,45 +483,114 @@ func updateNodesStatus(c client.Client, instance *computenodev1alpha1.ComputeNod
 				if err != nil {
 					return err
 				}
 			}
 		case "SchedulingDisabled":
-			// if drain has finished (pod completed) remove blocker-pod finalizer
-			drained, err := isNodeDrained(c, node.Name, instance)
-			if err != nil {
+			log.Info(fmt.Sprintf("Previous Status SchedulingDisabled: %s", node.Name))
+			// if pre draining has finished (job completed) remove blocker-pod finalizer
+			drainedPre, err := isNodeDrained(c, node.Name, node.Name+"-drain-job-pre", instance)
+			if err != nil && errors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("Pre NodeDrain job errors.IsNotFound: %s", node.Name+"-drain-job-pre"))
+				ResyncPeriod = 30 * time.Second
+				return nil
+			} else if err != nil {
 				return err
 			}
-			if drained {
-				err = deleteBlockerPodFinalizer(c, node.Name, instance)
+
+			log.Info(fmt.Sprintf("Pre draining job: %s", drainedPre))
+			switch drainedPre {
+			case "succeeded":
+				// taint node that daemonset pods get stopped
+				nodes, err := kclient.CoreV1().Nodes().List(metav1.ListOptions{})
+
+				for _, updateNode := range nodes.Items {
+					log.Info("NodeDrainPre updateNode: " + updateNode.Name)
+					if updateNode.Name == node.Name {
+						log.Info("NodeDrainPre our node: " + updateNode.Name)
+
+						err = addToBeRemovedTaint(kclient, updateNode)
+						if err != nil {
+							return err
+						}
+					}
+				}
+
+				// wait until all pods in openstack namespace got shutdown
+				podsOnNode, err := getRunningReadyPodsOnNode(kclient, node.Name)
 				if err != nil {
 					return err
 				}
+				if podsOnNode > 0 {
+					log.Info(fmt.Sprintf("Still %s openstack pods in Running state: %s", podsOnNode, node.Name))
+					ResyncPeriod = 30 * time.Second
+					return nil
+				}
 
-				for i, nodeToDelete := range instance.Spec.NodesToDelete {
-					if nodeToDelete.Name == node.Name {
-						if len(instance.Spec.NodesToDelete) == 1 {
-							instance.Spec.NodesToDelete = []computenodev1alpha1.NodeToDelete{}
-						} else {
-							instance.Spec.NodesToDelete = removeNode(instance.Spec.NodesToDelete, i)
-						}
-						err = c.Update(context.TODO(), instance)
-						if err != nil {
-							return err
+				// trigger node drain post tasks (new pod)
+				err = triggerNodeDrain(c, node.Name, instance, false, openstackClientAdmin, openstackClient)
+				if err != nil {
+					return err
+				}
+				log.Info("Node drain POST job created!")
+				// if post draining has finished (job completed) remove the pod
+				drainedPost, err := isNodeDrained(c, node.Name, node.Name+"-drain-job-post", instance)
+				if err != nil && errors.IsNotFound(err) {
+					log.Error(err, fmt.Sprintf("Post NodeDrain job errors.IsNotFound: %s", node.Name+"-drain-job-post"))
+					ResyncPeriod = 30 * time.Second
+					return nil
+				} else if err != nil {
+					return err
+				}
+				log.Info(fmt.Sprintf("NodeDrainPost status: %v ", drainedPost))
+				switch drainedPost {
+				case "succeeded":
```

Comment:
don't we need to remove the Jobs here?

Reply:
with the used ResyncPeriod instead of watching the jobs the final status is None. Now with watching the jobs we can move it here.